docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,902 @@
+# `MeshTrafficRoute`
+
+- Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4743
+Other issues covered:
+
+- https://github.com/kumahq/kuma/issues/4691
+- https://github.com/kumahq/kuma/issues/4690
+
+## Context and Problem Statement
+
+Now that we have the [new policy matching resources](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md), it's time to create one for traffic routing.
+We currently have both `TrafficRoute` and `MeshGatewayRoute`.
+
+This MADR does not address Gateway API integration per se but does consider Gateway
+API's `HTTPRoute`, both as inspiration as well as to make sure we don't block
+integration.
+
+## Decision Drivers
+
+- `MeshGatewayRoute` and `TrafficRoute` have a large overlap in functionality
+  and the use cases for routing from a gateway and between services are similar
+- Gateway API has different resources per protocol
+
+## Considered Options
+
+- Two resources: `MeshGatewayRoute` and `MeshTrafficRoute`
+- Unified resource for services and gateway: `MeshTrafficRoute`
+- Unified resources but one per protocol, e.g. `MeshHTTPRoute`, `MeshTCPRoute`
+- Two resources: `MeshGatewayRoute` and `MeshTrafficRoute`, which both point to
+  a `HTTPRouteRules` resource
+
+## Decision Outcome
+
+We decided to create two resources `MeshGatewayHTTPRoute` and `MeshHTTPRoute`.
+They have the exact same structure for routing rules but the way they refer to
+targets is different.
+
+Each future protocol will get its own resource:
+
+- `MeshHTTPRoute`
+- `MeshTCPRoute`
+
+which can be extended with others like GRPC.
+
+### `MeshHTTPRoute`
+
+The new resource routes and alters requests from one service to another
+depending on where the request coming from and where it's going to.
+Routes end up in the client-side Envoy config so the source of the affected requests will
+be under `spec.targetRef`. The destination we can select in `spec.to`.
+
+```yaml
+spec:
+ targetRef: ... # from where requests are coming
```

Comment:
Now this targetRef selects proxies that will receive a conf, right? If so, the comment is slightly misleading

Reply:
How is it misleading?

Reply:
Requests are always coming from the app that's running next to the sidecar. TargetRef selects not "from where requests are coming" but "group of proxies that will get routes on their outbound listeners".

Reply:
I explained that just around this code snippet. The from/to idea is an additional explanation that doesn't depend on Envoy technicals to understand which requests in the mesh are going to be affected by this route.